cdap-hbase-compat-1.1/src/main/java/co/cask/cdap/data2/transaction/queue/coprocessor/hbase11/HBaseQueueRegionObserver.java

```diff
@@ -188,11 +230,14 @@ private TransactionStateCache getTxStateCache() {
     private long rowsEvicted = 0;
     // couldn't be evicted due to incomplete view of row
     private long skippedIncomplete = 0;
+    private boolean invalidTxData;
 
-    private EvictionInternalScanner(String action, RegionCoprocessorEnvironment env, InternalScanner scanner) {
+    private EvictionInternalScanner(String action, RegionCoprocessorEnvironment env, InternalScanner scanner,
+                                    @Nullable TransactionVisibilityState state) {
```

Comment:
Just checked that we handle null `TransactionVisibilityState` in `CompactionState`. In that case we don't write compaction state.